tests/mocha/test_helpers.js

```diff
@@ -397,17 +413,57 @@ function assertNthCallEventArgEquals(spy, n, instanceType, expectedProperties,
   assertXmlProperties_(eventArg, xmlProperties);
 }
 
-function defineStackBlock() {
-  Blockly.defineBlocksWithJsonArray([{
+/**
+ * Adds message to shared cleanup object so that it is cleaned from
+ *    Blockly.Messages global in sharedTestTeardown.
+ * @param {!Object} sharedCleanupObj The shared cleanup object created in
+ *    sharedTestSetup.
+ * @param {string} message The message to add to shared cleanup object.
+ */
+function addMessageToCleanup(sharedCleanupObj, message) {
+  sharedCleanupObj.messagesCleanup_.push(message)
+}
+
+/**
+ * Adds block type to shared cleanup object so that it is cleaned from
+ *    Blockly.Blocks global in sharedTestTeardown.
+ * @param {!Object} sharedCleanupObj The shared cleanup object created in
+ *    sharedTestSetup.
+ * @param {string} blockType The block type to add to shared cleanup object.
+ */
+function addBlockTypeToCleanup(sharedCleanupObj, blockType) {
+  sharedCleanupObj.blockTypesCleanup_.push(blockType)
+}
+
+/**
+ * Calls Blockly.defineBlocksWithJsonArray with provided jsonArray and adds
+ *    block types to shared cleanup object so that they are cleaned from
+ *    Blockly.Blocks global in sharedTestTeardown.
+ * @param {!Object} sharedCleanupObj The shared cleanup object created in
+ *    sharedTestSetup.
+ * @param {!Array.<!Object>} jsonArray An array of JSON block definitions to
+ *    use in Blockly.defineBlocksWithJsonArray call.
+ */
+function defineBlocksWithJsonArrayWithCleanup(sharedCleanupObj, jsonArray) {
+  jsonArray.forEach((jsonBlock) => {
+    if (jsonBlock) {
+      addBlockTypeToCleanup(sharedCleanupObj, jsonBlock['type']);
+    }
+  });
+  Blockly.defineBlocksWithJsonArray(jsonArray);
+}
```

Comment:
Stubbing would be reasonable. In this case as well, the JSON tests should not use the stub, but should call the original function.

Is there a central list of stubs that are automatically set up by doing shared test setup?

Reply:
To clarify, the stub would still call the original function.
```
sinon.stub(Blockly, 'defineBlocksWithJsonArray(jsonArray).callsFake((jsonArray) => {
  if (jsonArray) {
    jsonArray.forEach((jsonBlock) => {
      if (jsonBlock) {
        addBlockTypeToCleanup(sharedCleanupObj, jsonBlock['type']);
      }
    });
  }
  // Calls original method.
  stub.wrappedMethod.call(this, ...arguments);
});
```
It seems fine to me to use this stub for JSON tests since it calls the original either way. Otherwise, I would need the stub and also the `defineBlocksWithJsonArrayWithCleanup`. Is there a reason this wouldn't be good?

The shared setup only defines stubs for `Events.fire` and the `clock`, which are accessible with `this.eventsFireStub` and `this.clock`. There isn't a need for defining a list of stubs that I know of, because calling `sinon.restore` will restore all stubs defined on `sinon` (which is the only reason I can think of where you'd want to know all the stubs defined.

Reply:
Using the stub for JSON tests seems fine if it's calling the original and is a thin wrapper.

When I say a list of stubs, I mean a human-readable list, because you may need to know what stubs you have to know if you're actually testing the code you expect to test.

Reply:
I've updated the JSDoc for sharedTestSetup to clarify what stubs are created.